DataDog/puppet-datadog-agent
# [CECO-1729] Installer support

### What does this PR do?

Adds support for the Datadog installer within Puppet, including a kitchen test

### Describe your test plan

Covered by kitchen E2E test and manually tested:

1. Run a Ubuntu vm as part of a Puppet cluster as agent
2. Install the installer using the following node manifest:
```puppet
class { "datadog_agent":
  api_key => "REPLACE_ME",
  datadog_installer_enabled => true,
  apm_instrumentation_enabled => "host",
  apm_instrumentation_libraries => ["java:1", "python:2"],
  remote_updates => "true",
  log_level => "debug",
  agent_extra_options => {
    hostname_trust_uts_namespace => true,
  },
}
```
3. Ensure the installer is installed and that it added your desired packages:
```shell
/usr/bin/datadog-installer is-installed datadog-apm-library-java
# should exit code 0
/usr/bin/datadog-installer is-installed datadog-apm-library-python
# should exit code 0
/usr/bin/datadog-installer is-installed datadog-agent
# should exit code 0 if remote_updates is true, 10 otherwise
```
4. Verify the Agent config file is correct, includes your additional options, `remote_updates` with the value you set
